src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,376 @@
+        /// Gets or Sets Collectors for VSTest run.
+        /// </summary>
+        public string[] VSTestCollect { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test failure.
+        /// </summary>
+        public string VSTestBlame { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test crash.
+        /// </summary>
+        public string VSTestBlameCrash { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Dumptype used for crash source blame.
+        /// </summary>
+        public string VSTestBlameCrashDumpType { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test crash even if test pass.
+        /// </summary>
+        public string VSTestBlameCrashCollectAlways { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test hang.
+        /// </summary>
+        public string VSTestBlameHang { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Dumptype used for hang source blame.
+        /// </summary>
+        public string VSTestBlameHangDumpType { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Time out for hang source blame.
+        /// </summary>
+        public string VSTestBlameHangTimeout { get; set; }
+
+        /// <summary>
+        /// Gets or Sets The directory path where trace data collector is.
+        /// </summary>
+        public string VSTestTraceDataCollectorDirectoryPath { get; set; }
+
+        /// <summary>
+        /// Gets or Sets disabling Microsoft logo while running test through VSTest.
+        /// </summary>
+        public string VSTestNoLogo { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Test artifacts processing mode which is applicable for .NET 5.0 or later versions.
+        /// </summary>
+        public string VSTestArtifactsProcessingMode { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Correlation Id of test session.
+        /// </summary>
+        public string VSTestSessionCorrelationId { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Runner version of VSTest.
+        /// </summary>
+        [Required]
+        public string VSTestRunnerVersion { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Path to nuget package cache.
+        /// </summary>
+        [Required]
+        public string NugetPath { get; set; }
+
+        protected override string ToolName => "vstest.console.exe";
+
+        protected override MessageImportance StandardOutputLoggingImportance => MessageImportance.High;
+
+        protected override MessageImportance StandardErrorLoggingImportance => MessageImportance.High;
+
+        /// <inheritdoc/>
+        protected override string GenerateFullPathToTool()
+        {
+#if NET6_0_OR_GREATER
+            return $@"{NugetPath}microsoft.testplatform\{VSTestRunnerVersion}\tools\net6.0\Common7\IDE\Extensions\TestPlatform\vstest.console.exe";
```

Comment:
The net6.0 directory doesn't exist in the Microsoft.TestPlatform package:

![image](https://github.com/microsoft/MSBuildSdks/assets/6445614/3077e44b-a83c-4f75-be57-8e813252bcdd)